### PR TITLE
Build Debos test containers

### PR DESCRIPTION
**Pull request by obbardc**
Currently the Debos CI pipeline builds the container each run, which can cause slow developer feedback. Build Debos build containers such that we can run a test build and the unit tests without having to build a container and hence decrease the time taken for a developer to get feedback.

### CODE REVIEW

File: .github/workflows/container.yaml [hunk 1]
```diff
@@ -40,6 +40,7 @@ jobs:
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/${{ matrix.variant }}
           tags: |
+            "type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}"
```
Comment (sjoerdsimons) [inline .github/workflows/container.yaml:53]: latest gets generated by event=tag by default; i've not bothered with an explicit latest tag on purpose

File: .github/workflows/container.yaml [hunk 2]
```diff
@@ -22,8 +22,17 @@ jobs:
       matrix:
         variant:
           - fakemachine-arch
-          - fakemachine-bullseye
-          - fakemachine-bookworm
+          - fakemachine-debian-bullseye
+          - fakemachine-debian-bookworm
```
Comment (sjoerdsimons) [inline .github/workflows/container.yaml:26]: don't chagn the variant names

File: .github/workflows/container.yaml [hunk 3]
```diff
@@ -50,7 +61,68 @@ jobs:
         uses: docker/build-push-action@v2
         with:
           context: .
-          file: Dockerfile.${{ matrix.variant }}
+          file: Dockerfile.${{ matrix.dockerfile }}
+          push: true
```
Comment (sjoerdsimons) [inline .github/workflows/container.yaml:65]: now you're pushing from a PR branch to the organisations package list; please avoid that; that's why theprevoius one only pushed on !PR's

File: .github/workflows/container.yaml (hunk 2 above)
Comment (obbardc) [reply .github/workflows/container.yaml:26]: right, i thought calling them `debian-*` would be clearer, but perhaps not :-)

File: Dockerfile.debos-arch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+ARG BASE_REGISTRY=ghcr.io/go-debos/test-containers
+ARG BASE_IMAGE=fakemachine-arch
+ARG BASE_TAG=latest
+FROM $BASE_REGISTRY/$BASE_IMAGE:$BASE_TAG
+
+# Bits needed to run debos test suite
+RUN pacman -Syu --noconfirm dpkg \
+                            unzip
```
Comment (evelikov) [inline Dockerfile.debos-arch:8]: Side note: We should explore switching to libarchive and it's tools - bsdtar, etc
 - single tool to interact with - supporting dpkg, zip, rar, nearly every tar permutation...
 - more widely available due to its BSD license
 - always present and working - as Arch is concerned - our package manager (pacman) depends on libarchive

Comment (obbardc) [reply Dockerfile.debos-arch:8]: right - can you open an issue on debos so that doesn't get lost ? :-)

Comment (evelikov) [reply Dockerfile.debos-arch:8]: Good call - forwarded to https://github.com/go-debos/debos/issues/316

File: .github/workflows/container.yaml (hunk 3 above)
Comment (obbardc) [reply .github/workflows/container.yaml]: fixed by caching the intermediate artifacts

**Comment by evelikov** [comment]
Looks good, thanks for getting back. Personally I would have kept the cosmetics and functional changes separate in the last commit, but that's a nitpick.

**Review by sjoerdsimons — APPROVED** [review]

**Pull request merged**